File: publisher/publisher.php
<?php
/**
 * Simple message sender to an AMQP server
 *
 * @author Osman Yuksel < yuxel |ET | sonsuzdongu |DOT| com >
 */

//path to your amqp-php library
require_once('../amqp.inc');


class sendMailToQueue {
  //set configs
  function __construct() {
    $this->rabbitMQConf = array("host"=>"localhost",
                                "port"=>"5672",
                                "user"=>"guest",
                                "pass"=>"guest",
                                "virtualHost"=>"/");


    $this->queueConf = array("exchange"=>"bulletin");
  }

  //send  $text $to someone with a $subject
  function send($to,$subject,$text) {
    try{
      $conn = new AMQPConnection($this->rabbitMQConf['host'], 
                                 $this->rabbitMQConf['port'],
                                 $this->rabbitMQConf['user'],
                                 $this->rabbitMQConf['pass']);

      $channel = $conn->channel();

      $channel->access_request($this->rabbitMQConf['virtualHost'], false, false, true, true);

      $mailData = array("to"=>$to,
                        "subject"=>$subject,
                        "text"=>$text);

      $mailDataToJson = json_encode($mailData);
 
      $msg = new AMQPMessage($mailDataToJson, array('content_type' => 'text/plain'));
      $channel->basic_publish($msg, $this->queueConf['exchange']);

      $channel->close();
      $conn->close();
      
      return true;
    }
    catch(Exception $e) {
      echo "Something went wrong ".$e->getMessage();
    }

  }
}


$mailSender = new sendMailToQueue();

$to = "foo@bar.com";
$subject = "This is a text message";
$text = "I show you how deep the rabbit-hole goes";

if ( $mailSender->send($to, $subject, $text ) ) {
   echo "Mail sent to queue"; 
}

File: publisher/publisher.py
""" 
Simple message sender to an AMQP server

@author Osman Yuksel < yuxel |ET | sonsuzdongu |DOT| com >
"""

from amqplib import client_0_8 as amqp
import simplejson as json # to encode message into json


class sendMailToQueue:

  #configs
  def __init__(self):
    self.rabbitMQConf = {"hostPort" : "localhost:5672",
                         "user" : "guest",
                         "pass" : "guest",
                         "virtualHost" : "/"
    }

    self.queueConf = {"exchange" : "bulletin"}


  #send 'text' 'to' someone with a 'subject'
  def send(self, to, subject, text):
    mailData = {"to" : to,
                "subject" : subject,
                "text" : text
    }
    
    # encode data into JSON
    mailDataToJson = json.dumps ( mailData )

    try:
      conn = amqp.Connection(self.rabbitMQConf['hostPort'],
                             self.rabbitMQConf['user'],
                             self.rabbitMQConf['pass'],
                             virtualHost = self.rabbitMQConf['virtualHost'],
                             insist=False)
      chan = conn.channel()

      msg = amqp.Message(mailDataToJson)
      msg.properties["delivery_mode"] = 2 # message is persistent
      chan.basic_publish(msg,exchange=self.queueConf['exchange'])

      chan.close()
      conn.close()

    except Exception, error :
      print "Something went wrong : ",error


mailSender = sendMailToQueue()
mailSender.send("foo@bar.com",
                "This is text message",
                "I show you how deep the rabbit-hole goes")

